src/Truck.java

```diff
@@ -1,17 +1,19 @@
-public class Truck {
-
-    public String modelName;
-    public int wheelsCount;
-
-    public void updateTyre() {
-        System.out.println("Меняем покрышку");
+public class Truck extends Vehicle {
+    public Truck(String modelName, int wheelsCount) {
+        super(modelName, wheelsCount);
     }
 
-    public void checkEngine() {
-        System.out.println("Проверяем двигатель");
+    public void check() {
+        this.updateTyre();
+        this.checkEngine();
+        this.checkTrailer();
     }
 
-    public void checkTrailer() {
+    private void checkTrailer() {
         System.out.println("Проверяем прицеп");
     }
+
+    private void checkEngine() {
+        System.out.println("Проверяем двигатель");
```

Comment:
```suggestion
        System.out.println("Проверяем двигатель грузовика");
```
метод смены колеса тоже по-хорошему надо переопределить во всех 3 классах, потому что это разные реализации под каждый вид ТС